ambry-router/src/main/java/com/github/ambry/router/PutOperation.java

```diff
@@ -290,6 +297,8 @@ private PutOperation(RouterConfig routerConfig, NonBlockingRouterMetrics routerM
     metadataPutChunk = new MetadataPutChunk();
     chunkFillerChannel = new ByteBufferAsyncWritableChannel(writableChannelEventListener);
     isEncryptionEnabled = passedInBlobProperties.isEncrypted();
+    restRequest = options.getRestRequest();
```

Comment:
Nit: since `makeTouchMessage` is only called in the constructor, you could pass restRequest in as a makeTouchMessage argument instead of making it a member variable.